driver-legacy/src/main/com/mongodb/DBCursor.java

```diff
@@ -582,8 +582,12 @@ public DBCursor skip(final int numberOfElements) {
 
     @Override
     public long getCursorId() {
-        if (cursor != null && cursor.getServerCursor() != null) {
-            return cursor.getServerCursor().getId();
+        if (cursor != null) {
```

Comment:
SpotBugs found this from the implicit NonNull annotation on MongoCursor#getServerCursor return value